core/domain/user_query_services.py

```diff
@@ -47,20 +154,31 @@ def save_new_query_model(
             explorations edited by user.
 
     Returns:
-        query_id: str. ID of the UserQueryModel instance.
+        UserQuery. The newly created UserQuery.
```

Comment:
Why not just return the ID, like before? In most places, only the ID is used, and it is typically better to pass smaller things around where possible.

Reply:
Most of the place are tests, but in the controller's code, it is actually useful to pass the whole object (saves us getting it in the same place).

Reply:
I think we still generally prefer to pass IDs around despite that -- there is a caching layer on top of the datastore which is supposed to make repeated reads of the same item fast. How about keeping to that convention throughout?

Reply:
Okay, I don't have a strong opinion about this, done.